### PR TITLE
[polaris.shopify.com refresh] Feat/add google analytics

### LINKED ISSUE #5999: [polaris.shopify.com refresh] Add google analytics
- Page views
- Search terms
- Other events (icon downloads etc)

### PR DESCRIPTION
**Pull request by chazdean**
### WHY are these changes introduced?

Fixes #5999 

### WHAT is this pull request doing?

Adds [google analytics](https://analytics.google.com/analytics/web/#/report-home/a49178120w143489507p148106035) to the website
Currently google analytics is set up to track all page views

Images below show the event being triggered with correct url and `pageview` property being sent to Google Analytics

Screen Shot 2022-06-17 at 8 58 29 AM

Screen Shot 2022-06-17 at 8 58 29 AM

 #### Next Steps
- Add Tracking ID environment var to production
- Track Search Terms
- Track Icons downloads

### CODE REVIEW

File: polaris.shopify.com/src/pages/_document.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                __html: `
+              window.dataLayer = window.dataLayer || [];
+              function gtag(){dataLayer.push(arguments);}
+              gtag('js', new Date());
+              gtag('config', '${process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS}', {
+                page_path: window.location.pathname,
+              });
+            `,
+              }}
+            />
```
Comment (gui-santos) [inline polaris.shopify.com/src/pages/_document.js]: Shouldn’t we use the [`next/script` component](https://nextjs.org/docs/basic-features/script) to implement these scripts?

In Next’s docs, I found this [google analytics example](https://github.com/vercel/next.js/blob/canary/examples/with-google-analytics/pages/_app.js) that uses `next/script` and it does everything in `_app`

Apart from these differences, they do the same thing that you are doing in the PR
so everything is great 🙌

File: polaris.shopify.com/lib/ga/index.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export const pageview = (url) => {
+    window.gtag('config', process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS, {
+      page_path: url,
+    })
+  }
+  
+export const event = ({ action, params }) => {
+    window.gtag('event', action, params)
+}
```
Comment (gui-santos) [inline polaris.shopify.com/lib/ga/index.js]: It looks like prettier is not formatting this file, right?
I mean, it seems that the indentation is a bit off, but I don't know why that would happen... do you?

**Review by gui-santos — CHANGES_REQUESTED** [review]
Hey Chaz!

This looks great, and it’s pretty much how they implement in Next.js official docs
I only got a couple of questions about using or not using `next/script`, but nothing that big

Cheers

File: polaris.shopify.com/src/pages/_document.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import Document, { Html, Head, Main, NextScript } from 'next/document'
+
+export default class MyDocument extends Document {
+    render() {
+      return (
+        <Html>
+          <Head>
+            {/* Global Site Tag (gtag.js) - Google Analytics */}
```
Comment (martenbjork) [inline polaris.shopify.com/src/pages/_document.js]: I think we can remove this comment :)

**Review by martenbjork — APPROVED** [review]

File: polaris.shopify.com/src/pages/_document.js (hunk 1 above)
Comment (chazdean) [reply polaris.shopify.com/src/pages/_document.js]: Thats a great example, thanks for sharing! I was able to make those changes and use the `next/script` directly in `_app`, it looks a lot cleaner! 👍🏾

File: polaris.shopify.com/src/pages/_document.js (hunk 3 above)
Comment (chazdean) [reply polaris.shopify.com/src/pages/_document.js]: Done!

File: polaris.shopify.com/lib/ga/index.js (hunk 2 above)
Comment (chazdean) [reply polaris.shopify.com/lib/ga/index.js]: Yeah I'm not sure why this didn't get caught by prettier. 🤔 I was able to correct the styling manually though!

**Review by gui-santos — APPROVED** [review]
🍬

File: polaris.shopify.com/lib/ga/index.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+export const GA_TRACKING_ID = process.env.NEXT_PUBLIC_GA_ID
```
Comment (romellogoodman) [inline polaris.shopify.com/lib/ga/index.js]: Could you possibly move this folder under src? This way all of the src code is located within the same root dir?

File: polaris.shopify.com/tsconfig.json [hunk 5]
```diff
@@ -20,6 +20,6 @@
     "noUnusedLocals": false,
     "noUnusedParameters": false
   },
-  "include": ["src", "**/*.json", "next-env.d.ts"],
+  "include": ["src", "**/*.json", "next-env.d.ts", "lib"],
```
Comment (romellogoodman) [inline polaris.shopify.com/tsconfig.json]: Moving under src would mean you don't need to add `lib` to this list

File: polaris.shopify.com/lib/ga/index.js (hunk 4 above)
Comment (chazdean) [reply polaris.shopify.com/lib/ga/index.js]: Done!

**Review by romellogoodman — APPROVED** [review]

**Pull request merged**